core/modules/system/system.module

```diff
@@ -3852,3 +3852,122 @@ function system_admin_paths() {
   );
   return $paths;
 }
+
+/**
+ * Sets up a form to save information automatically.
+ *
+ * This function adds a submit handler and a submit button to a form array. The
+ * submit function saves all the data in the form, using variable_set(), to
+ * variables named the same as the keys in the form array. Note that this means
+ * you should normally prefix your form array keys with your module name, so
+ * that they are unique when passed into variable_set().
+ *
+ * If you need to manipulate the data in a custom manner, you can either put
+ * your own submission handler in the form array before calling this function,
+ * or just use your own submission handler instead of calling this function.
+ *
+ * This function is deprecated in Backdrop because it saves to the old
+ * "variables" table. In Backdrop sites, developers are encouraged to save
+ * configuration using the config system. See config(), config_set(), and
+ * config_get() for information on saving into configuration.
```

Comment:
This docblock needs to be updated to reflect the current situation.

We should also add a `@since` line, such as:

```
@since 1.6.0 Restored and made compatible with configuration management.
```